flask_iiif/restful.py

```diff
@@ -171,12 +172,12 @@ def get(self, version, uuid, region, size, rotation, quality,
         # Trigger event after proccess the api request
         iiif_after_process_request.send(self, **api_after_request_parameters)
 
-        # Built the after request parameters
-        api_after_request_parameters = dict(
-            mimetype=mimetype,
-            image=to_serve
-        )
+        send_file_kwargs = {'mimetype': mimetype}
+        basename = secure_filename(request.args.get('dl', ''))
+        if basename:
+            send_file_kwargs.update(
+                as_attachment=True,
+                attachment_filename='{0}.{1}'.format(basename, image_format),
```

Comment:
this code does the extra `custom_filename`. However, it doesn't do the reasonable default filename when no custom filename is provided.